src/x509.rs

```diff
@@ -157,6 +157,74 @@ impl<S: BERDecodable + Integer, A: BERDecodable + SignatureAlgorithm, K: BERDeco
     }
 }
 
+/// BasicConstraints type as defined in RFC 5280.
+///
+/// It is recommended to use the `ca()` or `no_ca()` constructors instead of constructing
+/// instances of this type directly, to ensure the following statement from the RFC is enforced:
+/// "CAs MUST NOT include the `pathLenConstraint` field unless the `cA` boolean is asserted [...]".
+#[derive(Clone, Debug, Eq, PartialEq, Hash)]
+pub struct BasicConstraints {
+    pub ca: bool,
+    pub path_len_constraint: Option<u64>,
+}
+
+impl BasicConstraints {
+    pub fn ca(path_len_constraint: Option<u64>) -> Self {
+        BasicConstraints { ca: true, path_len_constraint }
+    }
+
+    pub fn no_ca() -> Self {
+        BasicConstraints { ca: false, path_len_constraint: None }
+    }
+
+    /// Returns the logical value of the `cA` field, as defined in RFC 5280.
+    pub fn is_ca(&self) -> bool {
+        self.ca
+    }
+
+    /// Returns the logical value of the `pathLenConstraint` field, as defined in RFC 5280.
+    ///
+    /// This method enforces the following statement from the RFC: "The `pathLenConstraint` field
+    /// is meaningful only if the `cA` boolean is asserted [...]". Hence, this method returns `None`
+    /// if the `cA` field is not asserted.
+    pub fn path_len_constraint(&self) -> Option<u64> {
+        if self.ca {
+            self.path_len_constraint
```

Comment:
I think we want to always return the `self.path_len_constraint`. Otherwise we still can't get the info from badly constructed certificates.